src/main.rs

```diff
@@ -73,15 +74,44 @@ fn remove(opt: cli::Opt, key: String) -> Result<()> {
     config.save(&opt.amber_yaml)
 }
 
-fn print(opt: cli::Opt) -> Result<()> {
+fn print(opt: cli::Opt, style: Option<cli::PrintStyle>) -> Result<()> {
     let config = config::Config::load(&opt.amber_yaml)?;
     let secret = config.load_secret_key()?;
     let pairs: Result<Vec<_>> = config.iter_secrets(&secret).collect();
     let mut pairs = pairs?;
     pairs.sort_by(|x, y| x.0.cmp(y.0));
-    pairs
-        .iter()
-        .for_each(|(key, value)| println!("export {}={:?}", key, value));
+
+    if let Some(style) = style {
+        let to_obj = |p: Vec<(&String, String)>| {
+            p.iter()
+                .map(|(k, v)| {
+                    let mut obj = HashMap::new();
+                    obj.insert("key", (*k).clone());
+                    obj.insert("value", v.clone());
+                    obj
+                })
+                .collect::<Vec<_>>()
+        };
+        match style {
+            cli::PrintStyle::SetEnv => pairs
+                .iter()
+                .for_each(|(key, value)| println!("export {}={:?}", key, value)),
+            cli::PrintStyle::Json => {
+                let secrets = to_obj(pairs);
+                let json = serde_json::to_string(&secrets).unwrap();
+                println!("{}", json)
+            }
+            cli::PrintStyle::Yaml => {
+                let secrets = to_obj(pairs);
+                let yaml = serde_yaml::to_string(&secrets).unwrap();
+                println!("{}", yaml)
+            }
+        }
+    } else {
+        pairs
```

Comment:
Personally I'd just default to `setenv` instead of creating a separate output format.

Reply:
This is what I meant in the issue, outputs no-unrelated text. It eases the following scripts a bit.